src/Widgets/StripeDialog.vala

```diff
@@ -0,0 +1,381 @@
+            overlay.add (icon);
+            overlay.add_overlay (overlay_icon);
+
+            var grid = new Gtk.Grid ();
+            grid.column_spacing = 12;
+            grid.row_spacing = 6;
+            grid.attach (overlay, 0, 0, 1, 2);
+            grid.attach (primary_label, 1, 0, 1, 1);
+            grid.attach (secondary_label, 1, 1, 1, 1);
+
+            error_layout.add (grid);
+
+            layouts.add_named (error_layout, "error");
+            layouts.show_all ();
+        }
+
+        layouts.set_visible_child_name ("error");
+        cancel_button.label = _("Pay Later");
+        pay_button.label = _("Retry");
+
+        cancel_button.sensitive = true;
+        pay_button.sensitive = true;
+    }
+
+    private void show_card_view () {
+        pay_button.label = _("Pay $%s.00").printf (amount.to_string ());
+        cancel_button.label = _("Cancel");
+
+        layouts.set_visible_child_name ("card");
+        is_payment_sensitive ();
+    }
+
+    private void validate (int entry, string new_text) {
+        try {
+            switch (entry) {
+            case 0:
+                var regex = new Regex ("""[a-z|0-9]+@[a-z|0-9]+\.[a-z]+""");
+                email_valid = regex.match (new_text);
+                break;
+            case 1:
+                card_valid = is_card_valid (new_text);
+                break;
+            case 2:
+                if (new_text.length != 4) {
+                    expiration_valid = false;
+                } else {
+                    var regex = new Regex ("""[0-9]{4}""");
+                    expiration_valid = regex.match (new_text);
+                }
+                break;
+            case 3:
+                var regex = new Regex ("""[0-9]{3,4}""");
+                cvc_valid = regex.match (new_text);
+                break;
+            }
+        } catch (Error e) {
+            warning (e.message);
+        }
+
+        is_payment_sensitive ();
+    }
+
+    private void is_payment_sensitive () {
+        if (email_valid && card_valid && expiration_valid && cvc_valid) {
+            pay_button.sensitive = true;
+        } else {
+            pay_button.sensitive = false;
+        }
+    }
+
+    private bool is_card_valid (string numbers) {
+        var char_count = numbers.char_count ();
+
+        if (char_count < 14) return false;
+
+        int hash = int.parse (numbers[char_count-1:char_count]);
+
+        int j = 1;
+        int sum = 0;
+        for (int i = char_count -1; i > 0; i--) {
+            var number = int.parse (numbers[i-1:i]);
+            if (j++ % 2 == 1) {
+                number = number * 2;
+                if (number > 9) {
+                    number = number - 9;
+                }
+            }
+
+            sum += number;
+        }
+
+        return 10 - (sum % 10) == hash;
+    }
+
+    private void on_response (Gtk.Dialog source, int response_id) {
+        switch (response_id) {
+        case Gtk.ResponseType.APPLY:
```

Comment:
I guess those `case` should have identation to distinguish them from `switch` statement?